testing/integ-test/src/crossVersionTest/groovy/org/gradle/integtests/TaskTransitiveSubclassingBinaryCompatibilityCrossVersionSpec.groovy

```diff
@@ -81,7 +81,9 @@ class TaskTransitiveSubclassingBinaryCompatibilityCrossVersionSpec extends Cross
                 @Override
                 JavaExec setArgs(List<String> args) {
                     println "args set: \$args"
-                    super.setArgs(args)
+                    // TODO: Fix for Gradle 9.0, we don't intercept super calls
+                    // was super.setArgs(args)
+                    (this as CustomBaseJavaExec).setArgs(args)
```

Comment:
❓ Is the plan to fix this only in 9.0? Can we revert this once this is fixed, preferably before 9.0?

- #29486

Reply:
Yes, maybe. I am not 100% sure we can fix such cases since, but we should take a look at it. I will add a comment to the issue you linked